### ISSUE TITLE
Why we have ~50k trials without registration_date?

### ISSUE THREAD

**Issue by vitorbaptista** [opentrials/opentrials#280]
This looks like a possible bug. It's not exclusive to a single registry, as can be seen in:

| PRIMARY_SOURCE_ID | COUNT |
| --- | --- |
| ictrp | 51,098 |
| gsk | 1,420 |
| pubmed | 442 |
| takeda | 409 |
| pfizer | 19 |
| **TOTAL** | **53,388** |

@roll thoughts?

**Comment by roll** [comment]
@vitorbaptista 
Per registries:
- ictrp - has not regular date format so we should improve `collecectors` logic to extract it
- gsk - I suppose it's related to https://github.com/opentrials/opentrials/issues/271
- pubmed - stub trials
- takeda, pfizer - we should check looking like not parsed dates

**Comment by vitorbaptista** [comment]
@nightsh Are you working on this?

**Comment by roll** [comment]
No one is assigned by we had conversation with @nightsh about ICTRP

**Comment by nightsh** [comment]
@vitorbaptista I tried and got stuck in an undocumented (yet) blocker. I will post an issue today specific to the ICTRP collector.

@roll I was able to reproduce the issue on a 2nd machine since then, gathered no new information. I'm filing an issue (a bit later on, I'm at a conference now).

Talk to you guys tomorrow a bit about this?

**Comment by roll** [comment]
To allow smart date parsing we need to change only on thing:

https://github.com/opentrials/collectors/blob/master/collectors/base/fields.py#L72-L83
(adding like `guess=True` flag)

**Comment by vitorbaptista** [comment]
I fixed ICTRP on https://github.com/opentrials/processors/issues/51, the other sources were OK.

I thought about fixing it on the `collectors` level, as there're not that many formats (2). However, after I started checking the `date_of_first_enrollment`, I changed my mind. There I found 7 different formats:

```python
'%d/%m/%Y'
'%d/%Y'
'%Y-%m-%d'
'%Y/%m/%d'
'%Y-%m-00'  # I guess this is a hack to refer to just a month/year
'%Y%m%d'
'%B %Y'
```
Also some invalid dates like `00/00/0000`, `08/00/2008`, `19/4/213`, etc. Having it in the `processor` level allow us to debug and improve the cleaning without needing to scrape again.
